src/pykeen/nn/node_piece.py

```diff
@@ -229,9 +245,8 @@ def __call__(self, edge_index: numpy.ndarray, known_anchors: numpy.ndarray) -> n
         if no_convergence:
             logger.warning(f"No covergence after {self.max_iter} iterations with epsilon={self.epsilon}.")
 
-        nodes = numpy.argpartition(x, max(x.size - self.num_anchors, 0))
-        unique_anchors = nodes[~numpy.isin(nodes, known_anchors)][-self.num_anchors:]
-        unique_anchors = numpy.concatenate([known_anchors, unique_anchors])
+        nodes = numpy.argsort(x)
```

Comment:
the argsort is also shared among both heuristics 😉

Reply:
I keep in mind the future `RandomSelection` module which won't need argsort, so I'd rather keep it out of the filtering function by now

Reply:
🤯 

alrighty 👍